File: src/main/java/eladkay/quaritum/common/block/chalk/EnumChalkColor.java
package eladkay.quaritum.common.block.chalk;

import eladkay.quaritum.common.block.flowers.BlockAnimusFlower;
import eladkay.quaritum.common.lib.LibNames;
import net.minecraft.util.IStringSerializable;

import java.util.Arrays;
import java.util.stream.Collectors;

public enum EnumChalkColor implements IStringSerializable {
    WHITE,
    ORANGE,
    MAGENTA,
    LIGHT_BLUE,
    YELLOW,
    LIME,
    PINK,
    GRAY,
    SILVER,
    CYAN,
    PURPLE,
    BLUE,
    BROWN,
    GREEN,
    RED,
    BLACK,
    TEMPEST;
    public static String[] vars = new String[EnumChalkColor.values().length];

    static {
        for (EnumChalkColor var : EnumChalkColor.values()) {
            vars[var.ordinal()] = LibNames.CHALK + String.join(",", Arrays.asList(var.getName().split("_")).stream().map(BlockAnimusFlower::capitalizeFirst).collect(Collectors.joining()));
            System.out.println(LibNames.CHALK + String.join(",", Arrays.asList(var.getName().split("_")).stream().map(BlockAnimusFlower::capitalizeFirst).collect(Collectors.joining())));
        }

    }


    public static EnumChalkColor of(int meta) {
        if (meta < 0 || meta > values().length) return null;
        return values()[meta];
    }

    @Override
    public String getName() {
        return toString().toLowerCase();
    }
}

File: src/main/java/eladkay/quaritum/common/block/chalk/EnumChalkColor.kt
package eladkay.quaritum.common.block.chalk

import eladkay.quaritum.common.block.flowers.BlockAnimusFlower
import eladkay.quaritum.common.lib.LibNames
import net.minecraft.util.IStringSerializable

enum class EnumChalkColor : IStringSerializable {
    WHITE,
    ORANGE,
    MAGENTA,
    LIGHT_BLUE,
    YELLOW,
    LIME,
    PINK,
    GRAY,
    SILVER,
    CYAN,
    PURPLE,
    BLUE,
    BROWN,
    GREEN,
    RED,
    BLACK,
    TEMPEST;

    override fun getName(): String {
        return toString().toLowerCase()
    }

    companion object {
        var vars = Array(EnumChalkColor.values().size) {
            val variant = EnumChalkColor.values()[it]
            LibNames.CHALK + variant.getName()
                    .split("_".toRegex())
                    .filter(String::isNotEmpty)
                    .map { BlockAnimusFlower.capitalizeFirst(it) }
                    .joinToString("")
        }


        fun of(meta: Int): EnumChalkColor? {
            if (meta < 0 || meta > values().size) return null
            return values()[meta]
        }
    }
}
